pkg/cmd/step/syntax/step_syntax_effective.go

```diff
@@ -420,6 +420,21 @@ func (o *StepSyntaxEffectiveOptions) createPipelineForKind(kind string, lifecycl
 		if err != nil {
 			return nil, errors.Wrapf(err, "Failed to generate pipeline from build pack")
 		}
+
+		// lets remove any environment variables defined in the container options from build packs/pod templates
+		// which are explicitly passed in to avoid possibly adding a value and valueFrom expression in the
+		// generated pod's container env vars
+		co := parsed.Options.ContainerOptions
+		for _, ev := range pipelineConfig.Env {
+			name := ev.Name
+			for i, e := range co.Env {
+				if e.Name == name {
+					co.Env = append(co.Env[0:i], co.Env[i+1:]...)
+					break
+				}
+			}
+		}
+		co.Env = RemoveDuplicateEnvVars(co.Env)
```

Comment:
Is this an additional step? This is has nothing to do with the removal between duplicated in container options and pipeline env, right?

Reply:
Yeah, I'm very confused at this point. =)

Reply:
Still not sure what the intend of ` RemoveDuplicateEnvVars` is as that stage.

Reply:
I'm confused too - removing duplicate values resulted in duplicates in the test case